### PR TITLE
fixing actionlib BUG in Ubuntu 20

### PR DESCRIPTION
**Pull request by marinagmoreira**
This fixes the BUG on Ubuntu 20:
[ERROR] [1626285660.006720060, 10.360000000] : (ros.Astrobee.actionlib) BUG: Got a transition to CommState [ACTIVE] when in SimpleGoalState [DONE]
[ WARN] [1626285661.000143376, 11.360000000] : (ros.Astrobee) Freeflyer action timed out on going active.
[ERROR] [1626285663.010712617, 13.368000000] : (ros.Astrobee.actionlib) BUG: Got a second transition to DONE

I tested the perching, cargo transport and reacquire position commands
The delay ended up being 0.1, same as poll, so I don't think it will influence the overall performance.
Let me know if there's anything we can improve to make this fix 'cleaner'

### CODE REVIEW

File: shared/ff_util/include/ff_util/ff_action.h [hunk 1]
```diff
@@ -214,6 +216,8 @@ class FreeFlyerActionClient {
       &FreeFlyerActionClient::DeadlineTimeoutCallback, this, true, false);
     timer_poll_ = nh->createTimer(to_poll_,
         &FreeFlyerActionClient::ConnectPollCallback, this, false, false);
+    timer_response_delay_ = nh->createTimer(to_response_delay_,
+        &FreeFlyerActionClient::ResultDelayCallback, this, false, false);
```
Comment (kbrowne15) [inline shared/ff_util/include/ff_util/ff_action.h:220]: If you make this a one shot timer, you don't have to worry about stopping the timer in the result delay callback.

File: shared/ff_util/include/ff_util/ff_action.h [hunk 2]
```diff
@@ -361,13 +365,29 @@ class FreeFlyerActionClient {
...
-      Complete(FreeFlyerActionState::ABORTED, result);
+      state_response_ = FreeFlyerActionState::ABORTED;
+
+    StartOptionalTimer(timer_response_delay_, to_response_delay_);
+  }
+
+  // Called when a result is received
+  void ResultDelayCallback(ros::TimerEvent const& event) {
+    // Feedback has been received after a result before. Stop tracking the goal
+    // so that this doesn't happen
```
Comment (kbrowne15) [inline shared/ff_util/include/ff_util/ff_action.h:385]: I think this comment can be removed. It doesn't match the code.

File: shared/ff_util/include/ff_util/ff_action.h (hunk 1 above)
Comment (marinagmoreira) [reply shared/ff_util/include/ff_util/ff_action.h]: done :)

File: shared/ff_util/include/ff_util/ff_action.h (hunk 2 above)
Comment (marinagmoreira) [reply shared/ff_util/include/ff_util/ff_action.h]: done :)

File: shared/ff_util/include/ff_util/ff_action.h (hunk 1 above)
Comment (marinagmoreira) [reply shared/ff_util/include/ff_util/ff_action.h]: I also tested it and it still works as expected

**Review by kbrowne15 — APPROVED** [review]

**Review by bcoltin — CHANGES_REQUESTED** [review]
Can you add a detailed comment explaining why the timer is needed? I am sure in a year everyone will forget... Then I think it looks good. :)

**Comment by bcoltin** [comment]
Sorry to be so picky... can you mention in the comment that it only shows up in ubuntu 20 / ros whatever?

**Comment by marinagmoreira** [comment]
@bcoltin Is it ok now?

**Review by bcoltin — APPROVED** [review]

**Comment by bcoltin** [comment]
Perfect thanks!

**Pull request merged**